privacy_evaluator/attacks/property_inference_attack.py

```diff
@@ -24,10 +25,28 @@ def __init__(
         :param dataset: dataset for training of shadow classifiers, test_data from dataset
         with concatenation [test_features, test_labels]
         """
+
+        if not (isinstance(dataset, tuple) and list(map(type, dataset)) == [np.ndarray, np.ndarray]):
+            raise TypeError("Dataset type should be of shape (np.ndarray, np.ndarray).")
+            
         self.dataset = dataset
-        # count of shadow training sets, must be eval
+
+        if not(isinstance(target_model, TensorFlowV2Classifier) or isinstance(target_model,PyTorchClassifier)):
+            raise TypeError("Target model must be of type Classifier.")
+
+        # count of shadow training sets, must be even
         self.amount_sets = 2
+        if self.amount_sets % 2 != 0 or self.amount_sets < 2:
+            raise ValueError("Number of shadow classifiers must be even and greater than 2.")
+
         self.input_shape = self.dataset[0][0].shape  # [32, 32, 3] for CIFAR10
+        self.classes = [0,1]
+        if len(self.classes) != 2:
+            raise ValueError("Currently attack only works with two classes.")
```

Comment:
I think we shouldn't hard-code self.classes but rather get it from the num_elements_per_classes dict . It also makes the if clause redundant since the length will always be == 2.

Reply:
We create the num_elements_per_classes according to the self.classes (before this was just hard-coded to be class 0 and 1). A list of classes should then also be an optional argument of the PIA and is only hard-coded right now, since we didn't take care of the whole hard-coding problem in general. I just had a look at your code for the hard-coding issue #136 and commented/reviewed what we meant. The if clause also wouldn't be redundant then.